edsnlp/data/huggingface_hub.py

```diff
@@ -0,0 +1,429 @@
+        Converter(s) to transform dataset dicts to Doc objects. Recommended
+        converters are `"hf_ner"` and `"hf_text"`. More information is available
+        in the [Converters](/data/converters) page.
+    shuffle: Literal['dataset', False] or bool
+        Whether to shuffle the dataset before yielding. If True or 'dataset', the
+        whole dataset will be materialized and shuffled (may be expensive).
+    seed: Optional[int]
+        Random seed for shuffling.
+    loop: bool
+        Whether to loop over the dataset indefinitely.
+    load_kwargs: dict
+        Dictionary of additional kwargs that will be passed to the
+        `datasets.load_dataset()` method.
+    kwargs: dict
+        Additional keyword arguments passed to the converter, these are
+        documented in the [Converters](/data/converters) page.
+
+    Returns
+    -------
+    Stream
+    """
+    load_kwargs = load_kwargs or {}
+
+    # If user passed a dataset identifier string, load it
+    ds = dataset
+    if isinstance(dataset, str):
+        ds = _load_hf_dataset_with_config(
+            dataset, split=split, name=name, load_kwargs=load_kwargs
+        )
+
+    else:
+        # If user passed a (split) name to select
+        if split is not None:
+            try:
+                ds = dataset[split]
+            except Exception:
+                raise ValueError(
+                    f"Cannot select split {split!r} from dataset {dataset!r}."
+                )
+    
+    # If no split was provided and the loaded dataset exposes multiple splits
+    # (e.g., a `DatasetDict`), pick the 'train' split by default and warn
+    # the user to be explicit.
+    if split is None and hasattr(ds, "keys") and "train" in ds.keys():
+        warnings.warn(
+            f"Dataset {dataset!r} contains multiple splits and no `split` "
+            f"was provided; using 'train' by default. Pass `split` to "
+            f"select another split.",
+            UserWarning,
+        )
+        ds = ds["train"]
+
+    if "hf_ner" in converter:
+        _validate_hf_ner_columns(list(ds.column_names), kwargs)
+    if "hf_text" in converter:
+        _validate_hf_text_columns(list(ds.column_names), kwargs)
+        
+    reader = HFDatasetReader(ds, shuffle=shuffle, seed=seed, loop=loop)
+    stream = Stream(reader=reader)
+
+    if converter:
+        conv, kwargs = get_dict2doc_converter(converter, kwargs)
+        stream = stream.map(conv, kwargs=kwargs)
+
+    return stream
+
+
+
+def _iter_from_stream(data_stream):
+    for item in data_stream.execute():
+        if isinstance(item, DatasetEndSentinel):
+            continue
+        if isinstance(item, (list, tuple)):
+            for rec in item:
+                yield rec
+        else:
+            yield item
+
+
+@registry.writers.register("huggingface_hub")
+def to_huggingface_hub(
+    data: Union[Any, Stream],
+    dataset_name: Optional[str] = None,
+    *,
+    push_to_hub: bool = False,
```

Comment:
```suggestion
```

Don't you think it simpler if we don't handle this ? Since it's quite easy for the user to to `data.to_huggingface_dataset().push_to_hub()`, and having the option to do it from the EDS-NLP function might lead to mistaken uploads (never say never), I'd remove the `push_to_hub` option

And if we want an EDS-NLP writer to actually write the the hub, we can turn this into a dedicated function (like the current name you gave it, ie `to_huggingface_hub`), distinct from the `data.to_huggingface_dataset()` function